api/envoy/config/listener/v3/listener_components.proto

```diff
@@ -165,9 +166,25 @@ message FilterChainMatch {
   repeated string application_protocols = 10;
 }
 
+// The configuration for on-demand filter chain. If this field is not empty in FilterChain message,
+// a filter chain will be built on-demand.
+// On-demand filter chains will be first built as placeholders when a listener is created in order to
+// speedup the warming up of the listener. The building and initialization of an on-demand filter chain
+// will be postponed to the arrival of new connections that require this filter chain.
+// The building process of on-demand filter chains will be similar to the one of normal filter chains.
+// However, a unique rebuilder with a unique init manager will be assigned to build each on-demand filter chain.
+// On the completion of building the on-demand filter chain, connections that require this filter chain will be retried.
+message OnDemandConfiguration {
+  // [#not-implemented-hide:] The timeout to wait for filter chain placeholders to complete rebuilding.
+  // If the timeout is reached, the filter chain rebuilding fails. Rebuilder send callbacks with failure signal
+  // so that end users will close all stored connections related to this filter chain.
```

Comment:
What is the end user in this context? Can you describe the end user behavior directly?

Reply:
Actually, end users who send connection requests do not know what happens here, their sockets will be connected when the rebuilding is completed. It is the worker's handler who stored the sockets instead of end users. End users will not be aware of the filter chain rebuilding. They only knows the connections are created successfully or not.